core/src/main/kotlin/ca/allanwang/kau/utils/ViewUtils.kt

```diff
@@ -67,6 +78,36 @@ fun View.snackbar(@StringRes textId: Int, duration: Int = Snackbar.LENGTH_LONG,
     setImageDrawable(icon.toDrawable(context, sizeDp = sizeDp, color = color, builder = builder))
 }
 
+val FloatingActionButton.isHidden
+    get() = !isShown
+
+fun FloatingActionButton.showIf(show:Boolean) = if (show) show() else hide()
+
+fun FloatingActionButton.hideIf(hide:Boolean) = if (hide) hide() else show()
+
+fun ViewGroup.inflate(layoutId:Int, attachToRoot:Boolean = false):View =
+        LayoutInflater.from(context).inflate(layoutId, this, attachToRoot)
+
+fun View.updateLeftMargin(left:Int) = updateMargins(left)
+
+fun View.updateTopMargin(top:Int) = updateMargins(-1, top)
+
+fun View.updateRightMargin(right:Int) = updateMargins(-1, -1, right)
+
+fun View.updateBottomMargin(bottom:Int) = updateMargins(-1, -1, -1, bottom)
+
+fun View.updateMargins(left:Int = -1, top:Int = -1, right:Int = -1, bottom:Int = -1) {
+    if (layoutParams is ViewGroup.MarginLayoutParams) {
+        val p = layoutParams as ViewGroup.MarginLayoutParams
+        val newLeft = if (left >= 0) left else p.leftMargin
+        val newTop = if (top >= 0) top else p.topMargin
+        val newRight = if (right >= 0) right else p.rightMargin
+        val newBottom = if (bottom >= 0) bottom else p.bottomMargin
+        p.setMargins(newLeft, newTop, newRight, newBottom)
+        requestLayout()
```

Comment:
Margins can actually be negative. I would rather taking in two values, the amount and a flag for which side and apply like that